### PR TITLE
feat(linkedlist): add first linked list job

### PR DESCRIPTION
**Pull request by priyasiddharth**
Reducing # of elements from 20 to 5 mainly to make --vac times reasonable
Currently 5 items takes 12s in dataflow mode, 30s in vac mode
10 items takes 5m in vac mode.

### CODE REVIEW

File: seahorn/jobs/array_list_back/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,4 @@
 add_executable(array_list_back
-  ${AWS_C_COMMON_ROOT}/source/array_list.c
```
Comment (agurfinkel) [inline seahorn/jobs/array_list_back/CMakeLists.txt:2]: mistake?

File: seahorn/jobs/linked_list_back/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# TODO: env allocation value from env/parent
+add_executable(linked_list_back
+  aws_linked_list_back_harness.c
+  ${SEA_LIB}/proof_allocators.c
+  ${SEA_LIB}/linked_list_helper.c)
+target_compile_definitions(linked_list_back
+  PUBLIC AWS_DEEP_CHECKS=0
```
Comment (agurfinkel) [inline seahorn/jobs/linked_list_back/CMakeLists.txt:7]: why no deep checks?

File: seahorn/jobs/linked_list_back/sea.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+verify_options:
+  # MAX_LINKED_LIST_ITEM_ALLOCATION to account for
+  # head and tail
+  bound: 7
```
Comment (agurfinkel) [inline seahorn/jobs/linked_list_back/sea.yaml:4]: should be set from cmake

File: seahorn/lib/linked_list_helper.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// This is the seahorn variant of
+// https://sourcegraph.com/github.com/awslabs/aws-c-common/-/blob/include/aws/common/linked_list.inl#L81
+// TODO: we can make this faster by getting rid of the loop
+// This can be accomplished by storing all the node addresses in an array
+// and asserting properties like
+// 1. Assert no inbetween node between [0,length] should be the tail
+// 2. Assert only the node at 'length' + 1 index should be the tail
+// 3. Assert that any node in [0, length] satisfies aws_linked_list_node_next_is_valid
+bool sea_aws_linked_list_is_valid(const struct aws_linked_list *list, size_t length) {
+    sassert(list &&
```
Comment (agurfinkel) [inline seahorn/lib/linked_list_helper.c:41]: strange to have asserts in a pure bool function

**Review by agurfinkel — CHANGES_REQUESTED** [review]
fantastic

**Comment by agurfinkel** [comment]
cbmc tests use 20 nodes in the linked list?

File: seahorn/jobs/linked_list_back/sea.yaml (hunk 3 above)
Comment (priyasiddharth) [reply seahorn/jobs/linked_list_back/sea.yaml:4]: done

File: seahorn/jobs/array_list_back/CMakeLists.txt (hunk 1 above)
Comment (priyasiddharth) [reply seahorn/jobs/array_list_back/CMakeLists.txt:2]: fixed

File: seahorn/lib/linked_list_helper.c (hunk 4 above)
Comment (priyasiddharth) [reply seahorn/lib/linked_list_helper.c:41]: bool will change to void.

File: seahorn/jobs/linked_list_back/CMakeLists.txt (hunk 2 above)
Comment (priyasiddharth) [reply seahorn/jobs/linked_list_back/CMakeLists.txt:9]: added comment with explaination.

**Pull request merged**